ocs_ci/ocs/platform_nodes.py

```diff
@@ -556,6 +538,142 @@ def restart_nodes_teardown(self):
+        node_list = []
+        node_cls = self.nodes_map[
+            f'{self.platform.upper()}{self.deployment_type.upper()}Node'
+        ]
+
+        if node_type.upper() == 'RHCOS':
+            workers_stacks = self.aws.get_worker_stacks()
+            logger.info(f"Existing worker stacks: {workers_stacks}")
+            existing_indexes = self.get_existing_indexes(workers_stacks)
+            logger.info(f"Existing indexes: {existing_indexes}")
+            slots_available = self.get_available_slots(existing_indexes, num_nodes)
+            logger.info(f"Available indexes: {slots_available}")
+            for slot in slots_available:
+                node_id = slot
+                node_list.append(node_cls(node_conf, node_type))
+                node_list[-1]._prepare_node(node_id)
+        elif node_type.upper() == 'RHEL':
+            rhel_workers = len(get_typed_worker_nodes('rhel'))
+            for i in range(num_nodes):
+                node_id = i + rhel_workers
+                node_list.append(node_cls(node_conf, node_type))
+                node_list[-1]._prepare_node(node_id)
+
+        # Make sure that csr is approved for all the nodes
+        # not making use of csr.py functions as aws rhcos has long
+        # delays for csr to appear
+        if node_type.upper() == 'RHCOS':
+            self.approve_all_nodes_csr(node_list)
+
+        return node_list
+
+    @retry(
+        (exceptions.PendingCSRException, exceptions.TimeoutExpiredError),
+        tries=4,
+        delay=10,
+        backoff=1
+    )
+    def approve_all_nodes_csr(self, node_list):
+        """
+        Make sure that all the newly added nodes are in approved csr state
+
+        Args:
+            node_list (list): of AWSUPINode/AWSIPINode objects
+
+        """
+        node_names = [
+            node.aws_instance_obj.private_dns_name for node in node_list
+        ]
+
+        sample = TimeoutSampler(
+            timeout=600, sleep=3, func=self.all_nodes_found,
+            node_names=node_names
+        )
+        if not sample.wait_for_func_status(result=True):
+            raise exceptions.PendingCSRException(
+                "All nodes csr not approved"
+            )
+
+    def all_nodes_found(self, node_names):
+        """
+        Relying on oc get nodes -o wide to confirm that
+        node is added to cluster
+
+        Args:
+            node_names (list): of node names as string
+
+        """
+        approve_pending_csr()
+        get_nodes_cmd = "get nodes -o wide"
+        oc_obj = ocp.OCP()
+        nodes_wide_out = oc_obj.exec_oc_cmd(
+            get_nodes_cmd, out_yaml_format=False
+        )
+        for line in nodes_wide_out.splitlines():
+            for node in node_names:
+                if node in line:
+                    node_names.remove(node)
+                    break
+        if node_names:
+            logger.warning("Some of the nodes have not appeared in nodes list")
+        return not node_names
```

Comment:
Return value not described in docstring.

Reply:
done